ledger/storage/record/record.go

```diff
@@ -21,15 +21,30 @@ import (
 )
 
 // TypeID encodes a record object type.
+//go:generate go run gen/type.go
 type TypeID uint32
 
 // TypeIDSize is a size of TypeID type.
 const TypeIDSize = 4
 
 // Record is base interface for all records.
 type Record interface {
-	// Type returns record type.
-	Type() TypeID
 	// WriteHashData writes record data to provided writer. This data is used to calculate record's hash.
 	WriteHashData(w io.Writer) (int, error)
 }
+
+func init() {
+	// Never change id constants.
+	register(10, new(GenesisRecord))
+	register(11, new(ChildRecord))
+	register(12, new(JetRecord))
+
+	register(20, new(RequestRecord))
+
+	register(30, new(ResultRecord))
+	register(31, new(TypeRecord))
+	register(32, new(CodeRecord))
+	register(33, new(ObjectActivateRecord))
+	register(34, new(ObjectAmendRecord))
+	register(35, new(DeactivationRecord))
+}
```

Comment:
I suggest to use hundreds or even thousands here.

Reply:
And also you should note here, that these are not magic numbers. This numbers haven't any meaning